File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/ResourceSpecifier.kt

package com.radixdlt.sargon.extensions

import com.radixdlt.sargon.ResourceAddress
import com.radixdlt.sargon.ResourceSpecifier
import com.radixdlt.sargon.resourceSpecifierGetAddress

val ResourceSpecifier.address: ResourceAddress
    get() = resourceSpecifierGetAddress(specifier = this)

File: crates/sargon-uniffi/src/wrapped_radix_engine_toolkit/low_level/execution_summary/resource_specifier/resource_specifier.rs
use crate::prelude::*;
use sargon::ResourceSpecifier as InternalResourceSpecifier;

#[derive(Clone, PartialEq, Eq, InternalConversionV2, uniffi::Enum)]
pub enum ResourceSpecifier {
    Fungible {
        resource_address: ResourceAddress,
        amount: Decimal,
    },
    NonFungible {
        resource_address: ResourceAddress,
        ids: Vec<NonFungibleLocalId>,
    },
}

#[uniffi::export]
pub fn new_resource_specifier_sample() -> ResourceSpecifier {
    InternalResourceSpecifier::sample().into()
}

#[uniffi::export]
pub fn new_resource_specifier_sample_other() -> ResourceSpecifier {
    InternalResourceSpecifier::sample_other().into()
}

#[uniffi::export]
pub fn resource_specifier_get_address(
    specifier: &ResourceSpecifier,
) -> ResourceAddress {
    specifier.into_internal().get_address().into()
}
